include/swift/AST/ModuleDependencies.h

```diff
@@ -109,6 +109,11 @@ class ModuleDependencyInfoStorageBase {
   /// The set of modules on which this module depends.
   std::vector<std::string> moduleImports;
 
+  /// The set of modules which constitute optional module
+  /// dependencies for this module, such as `@_implementationOnly`
+  /// or `internal` imports.
+  std::vector<std::string> optionalModuleImports;
```

Comment:
Just to clarify, a transitive `internal import` would either be ignored for a normal build or required for a `@testable import`. Only a transitive `@ _implementationOnly` is optional for a `@testable import`, or when loaded by the debugger.

Reply:
Does being required for a `@testable import` mean it is also then printed into the `.swiftinterface`? Or is this a case where the `.swiftinterface` does not contain an `import` even though the binary module treats it as required?

Reply:
The import is not printed in the swiftinterface, but right, it's still treated as required even if it's just in the swiftmodule.

Reply:
Cool, thank you for the clarification. 
Updated the PR to also query required dependencies from the adjacent binary module of a `@testable`-imported dependency. 👍🏼